### DISCUSSION TITLE
Passing slot data via schema

### DISCUSSION THREAD

**Discussion by benlind** [formkit/formkit#660] [Ideas]
Is it possible to pass slot content via a FormKit schema definition?

I am using the [multi-step plugin](https://formkit.com/plugins/multi-step) via schema definition and would like to add a submit button to the last step. The multi-step docs say that you have to pass a slot like so:

```vue
<template #stepNext>
  <FormKit type="submit" />
</template>
```

Is it possible to pass that slot via the schema? Here is my code:

```vue
<template>
  <FormKitSchema
    :schema="schema"
    :data="data"
  />
</template>

<script setup lang="ts">
const data = reactive({
  submitForm: async () => {
    await new Promise(r => setTimeout(r, 1000))
    alert('Submitted! 🎉')
  },
})

const schema = [
  {
    $formkit: 'form',
    actions: false,
    onSubmit: '$submitForm',
    children: [
      {
        $formkit: 'multi-step',
        children: [
          {
            $formkit: 'step',
            name: 'intro',
            label: 'Intro',
            children: 'Step 1 intro text',
          },

          {
            $formkit: 'step',
            name: 'general',
            label: 'General',
            children: [
              {
                $formkit: 'date',
                label: 'Choose a date',
                validation: 'required',
              },
              {
                // XXX: Can I pass this into the #stepNext slot?

                children: [
                  {
                    $formkit: 'submit',
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  },
]
</script>
```

**Comment by zpwparsons** [comment]
@benlind I'm struggling with the same thing. Did you find a solution?

**Reply by benlind** [reply]
No. We are avoiding the slot syntax for multi-steps at this point.

**Comment by fenilli** [comment]
Slots for defined components in Schema are a bit different, there is documentation about it here: https://formkit.com/essentials/schema#slots at the end called "FORMKIT INPUT SLOTS".

But basically it is by using raw values and `sectionsSchema`: https://formkit.com/essentials/schema#raw-values
